### ISSUE TITLE
Implement fine pixel tile scrolling

### ISSUE THREAD

**Issue by cspang1** [cspang1/JCAP#72]
Goal: ability to scroll background tiles smoothly horizontally, i.e. pixel-by-pixel. Could have system to allow for multiple different regions of different scroll rates. Primary goal however is to simply achieve fine scrolling.

Scrolling cannot be done on the CPU side - there's no way to de-conflict tile color palettes. So any fine scrolling must be done on the GPU side.

Horizontal scrolling will require some work - rendering tiles not on their boundaries results in needing to process an additional tile for each **long** of stored video data.

Planning out in scratchpad.

**Comment by cspang1** [comment]
Will need horizontal and vertical "position" tracking variables passed from CPU to control scrolling.

Hard-incremented on [this line](https://github.com/cspang1/JCAP/blob/master/Dev/Software/vga_render.spin#L342), as we will need the data for an extra tile during scrolling.

We instantiate a render cog internal tracker for which horizontal pixel we're rendering (**curpx**).

So the routine should operate as follows for horizontal fine scrolling (starting at the very beginning of a scanline):
1. We instantiate a variable **curpx** to track which physical pixel of the scanline we're currently rendering 
2. We retrieve a variable **h_pos** from the graphics resources to determine what pixel of the tile map is aligned with the left side of our screen
3. We determine what our initial tile is by performing **floor(h_pos/4) = floor(5/4) = 1**, and load that tile's **word** to be processed.
4. We then determine what specific pixel from the tile palette we're processing by performing **h_pos%4**, and load that pixel for rendering
5. When we proceed to the next pixel, we first check that we've entered a new tile by checking whether **curpx%4==0**. If so, we load the new tile and continue rendering pixels
6. We repeat steps 4 and 5 until we have a **long** worth of graphics data to store, with the option of either breaking out of the process when we hit that or letting the buffer overflow to save cycles on all of the checks that would be performed
